src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -880,40 +892,51 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			connection.options['azureAccountToken'] = tokenPromise.token;
 			return true;
 		}
+		let account = this.findAzureAccountFromProfile(accounts, connection);
+		if (account) {
+			this._logService.debug(`Getting security token for Azure account ${account.key.accountId}`);
+			if (account.isStale) {
+				this._logService.debug(`Account is stale - refreshing`);
+				try {
+					account = await this._accountManagementService.refreshAccount(account);
+				} catch (err) {
+					this._logService.info(`Exception refreshing stale account : ${toErrorMessage(err, true)}`);
+					// refreshAccount throws an error if the user cancels the dialog
+					return false;
+				}
+			}
+			const tenantId = connection.azureTenantId;
+			const token = await this._accountManagementService.getAccountSecurityToken(account, tenantId, azureResource);
+			this._logService.debug(`Got token for tenant ${token}`);
+			if (!token) {
+				this._logService.info(`No security tokens found for account`);
+			}
+			connection.options['azureAccountToken'] = token.token;
```

Comment:
This is going to throw if token is undefined (which you check for on line 911). Did you mean to return false there?

Reply:
Was this actually resolved?

Reply:
Oh, there's a return false above. False alarm 😄